src/nspawn/nspawn.c

```diff
@@ -4723,6 +4860,24 @@ static int run_container(
                         if (l != sizeof arg_uid_shift)
                                 return log_error_errno(SYNTHETIC_ERRNO(EIO), "Short write while writing UID shift.");
                 }
+
+                n_bind_user_uid = strv_length(arg_bind_user);
+                if (n_bind_user_uid > 0) {
+                        /* Right after the UID shift, we'll receive the list of UID mappings for the
+                         * --bind-user= logic. Always a quadruplet of payload and host UID + GID. */
+
+                        bind_user_uid = new(uid_t, n_bind_user_uid*4);
+                        if (!bind_user_uid)
+                                return log_oom();
+
+                        for (size_t i = 0; i < n_bind_user_uid; i++) {
+                                l = recv(uid_shift_socket_pair[0], bind_user_uid + i*4, sizeof(uid_t)*4, 0);
+                                if (l < 0)
+                                        return log_error_errno(errno, "Failed to read user UID map pair: %m");
+                                if (l != sizeof(uid_t)*4)
+                                        return log_error_errno(SYNTHETIC_ERRNO(EIO), "Short read while reading bind user UID pairs.");
```

Comment:
Hmm, maybe suppress this to log_debug when `l == 0`? I get the following output:
```console
$ sudo build/systemd-nspawn --resolv-conf=bind-host --link-journal=try-guest --private-network --network-veth --bind-user=zbyszek --private-users --private-users-ownership=map -M rawhide
...
UID of user 'zbyszek' to map is already in container UID range, refusing.
Short read while reading bind user UID pairs.
```